### ISSUE TITLE
multi line mode

### ISSUE THREAD

**Issue by smnorris** [jakob/Postico#131]
I did a quick search and didn't see this - sorry if this is a duplicate.

For editing sql another must-have feature is multi line mode.

My fav sql client Universal SQL Editor (http://mingsoftware.com/universalsqleditor, windows only unfortunately) does this with a simple Shift+Alt and arrow key select. Sublime Text multi line selections work well too but are probably more powerful than is really necessary.

For some reason Navicat has been really unstable lately so I've moved to using postico for browsing my db, it is looking pretty good.

**Comment by jakob** [comment]
Postico does support rectangular/column selection (drag mouse while holding alt key) and multiple selections (hold cmd key while selecting with mouse). This is built into the Cocoa text handling system, and most Mac apps can do that.

You can do a few things with multiple selections:
- delete them
- apply transformations (like uppercase/lowercase)
- copy them
- paste into them

Unfortunately Cocoa only supports a single insertion point, so you can't just start typing in multiple places at once. I guess this, together with the commands to modify multi line selections,  is the most useful feature in Sublime.

I assume that adding support for this would require some pretty substantial extension to the standard Cocoa text view; I'll keep it in mind, but I can't make any promises.

Just out of curiosity, what kinds of queries are you building with multi-line editing?

**Comment by smnorris** [comment]
Cool, I didn't know about that capability - thanks.

All kinds of queries - I'm not sure of the most common use.
Maybe change a common column name prefix in several adjacent lines.
Or add quotes around several lines of strings.
Or collapse lines of strings into a single comma separated list like in the
sublime home page example.

Anyway, simply using a text editor works fine too.

**Comment by jakob** [comment]
OK, thanks for your input. As I said, I'll keep these features in mind, but Postico will probably never reach the sophistication of a dedicated text editor.

**Comment by ear** [comment]
Trying to follow the spirit of your comment: have you considered adding support for query buffer editing similar to what `psql` provides? In particular the `\e`and `\i` commands.

**Comment by jakob** [comment]
Yeah, I've been thinking about ways to edit queries in an external editor, but I'm not sure how to best design the UI for that.

However, long term I'd like to implement something along the lines of #18. Then the query would just be a file on disk, and you could Cmd-Tab to an editor of your choice, then Cmd-Tab back and execute it.

**Comment by mrkcohen** [comment]
Still no chance in 2019 for multiline mode? Just thought I'd ask 🙃

**Comment by Ashtonian** [comment]
another +1 for multi line support for select similar, great for data editing

**Comment by VladimirDomnich** [comment]
Hello @jakob !
Thank you for the wonderful tool! I also often stumble upon lack of multi-cursor. My typical case is to wrap id-s placed on several lines into quotes so that I can run query by `IN` for them.

From this discussion I see that having multi-cursor is not quite feasible. But what about new transformations:
1. One transformation would convert selected lines into string literals (basically appending character ' in the beginning and end of each line).
2. Another transformation would put a comma in the end of each selected line.

Would something like this be possible?
